File: JAVA/349. Intersection of Two Arrays.java
class Solution {
    public int[] intersection(int[] n1, int[] n2) {
        Set<Integer> a = new HashSet<>();
        Set<Integer> b = new HashSet<>();
        List<Integer> x = new ArrayList<>();
        for (int num : n1) a.add(num);
        for (int num : n2) b.add(num);
        for (int i : a) {
            if (b.contains(i)) {
                x.add(i);
            }
        }
        int[] result = new int[x.size()];
        for (int i = 0; i < x.size(); i++) {
            result[i] = x.get(i);
        }        
        return result;
    }
}

File: C++/349. Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& n1, vector<int>& n2) {
        set<int>a(n1.begin(),n1.end());
        set<int>b(n2.begin(),n2.end());
        vector<int>x;
        for(int i:a){
            for(int j:b){
                if(i==j){
                    x.push_back(i);
                    break;
                }
            }
        }
        return x;
    }
};